include/plog/Util.h

```diff
@@ -189,18 +189,25 @@ namespace plog
 #endif
         }
 
-#ifdef _WIN32
+#if defined(_MSC_VER) && _MSC_VER <= 1600
+#   define va_copy(d,s) ((d)=(s)) // there is no va_copy on Visual Studio 2010
+#endif
+
+#ifndef __STDC_SECURE_LIB__
+#   define vsnprintf_s(buf,cnt,max,fmt,ap) vsnprintf((buf),(cnt),(fmt),(ap));
+#endif
+
+#ifndef _GNU_SOURCE
     inline int vasprintf(char** strp, const char* format, va_list ap)
     {
-#if defined(__BORLANDC__)
-        int charCount = 0x1000; // there is no _vscprintf on Borland/Embarcadero
-#else
-        int charCount = _vscprintf(format, ap);
+        va_list ap_copy;
+        va_copy(ap_copy, ap);
+        int charCount = vsnprintf_s(NULL, 0, static_cast<size_t>(-1), format, ap_copy);
```

Comment:
It's cool that `vsnprintf` returns number of characters. The documentation is vague about this point. Nice!